src/back/msl/writer.rs

```diff
@@ -887,6 +887,11 @@ impl<W: Write> Writer<W> {
             } => {
                 use crate::MathFunction as Mf;
 
+                let scalar_argument = match context.resolve_type(arg) {
```

Comment:
this should have "*", since otherwise you are matching a reference to the type

Reply:
Oh interesting! I've never had to do that before.

Reply:
yes, because we force-enable this requirement on clippy, and by default Rust tries to be too smart and derefefence for you. We don't want to have it too smart, we value explicitness more.